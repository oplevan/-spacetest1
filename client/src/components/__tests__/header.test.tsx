import { render, screen } from "@testing-library/react";
import Header from "../header";

describe("Header component", () => {
  test("renders header correctly", () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });

  test("renders with correct navigation links", () => {
    render(<Header />);

    const homeLink = screen.getByText("Home");

    expect(homeLink).toBeInTheDocument();
    expect(homeLink.getAttribute("href")).toBe("/");
  });

  test("renders navigation links with hover styles", () => {
    render(<Header />);

    const homeLink = screen.getByText("Home");

    expect(homeLink).toHaveClass("hover:text-blue-700");
    expect(homeLink).toHaveClass("hover:underline");
  });
});
