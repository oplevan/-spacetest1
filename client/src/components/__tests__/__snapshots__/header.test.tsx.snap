// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders header correctly 1`] = `
<div>
  <div
    class="py-3 shadow-md mb-4 lg:mb-10"
  >
    <nav
      class="container flex gap-4"
    >
      <a
        class="hover:text-blue-700 hover:underline"
        href="/"
      >
        Home
      </a>
    </nav>
  </div>
</div>
`;
